java-checks/src/main/java/org/sonar/java/checks/regex/RedosCheck.java

```diff
@@ -181,13 +211,16 @@ private void checkForOverlappingRepetitions(RepetitionTree tree) {
           if (reachabilityChecker.canReach(repetition, tree)) {
             SubAutomaton auto1 = new SubAutomaton(repetition.getElement(), tree.continuation(), false);
             SubAutomaton auto2 = new SubAutomaton(repetition.continuation(), tree.continuation(), false);
+            intersectionChecker.clearCache();
```

Comment:
Not sure to understand why we need to clear the cache here, could you explain this change?

Reply:
Damn, I forgot to remove that line. I added it while debugging when `intersects` kept returning different results than what it clearly should and what it did return in self-contained tests. So I suspected a caching bug (which turned out to be the case) and added the `clearCache` to verify. The issue went away, so that confirmed the caching bug. Then I (thought I) fixed the caching bug, but forgot to remove the `clearCache` before testing. Turns out the bug was only partially fixed and I'll have to investigate further :frowning_face:

Reply:
Turns out the part that wasn't fully fixed was due to another bug that I fixed in the PR that Alban merged earlier. So rebasing on master fixed the issue. :smile: